test/0.8.9/withdrawal-queue.test.js

```diff
@@ -719,77 +797,38 @@ contract('WithdrawalQueue', ([owner, stranger, daoAgent, user, pauser, resumer,
         })
         totalDistributedEth.iadd(bn(batch.ethToLock))
       }
-      const id = await withdrawalQueue.getLastRequestId()
-      await withdrawalQueue.finalize([id], await currentRate(), { from: steth.address, value: total })
       for (let index = 0; index < requestIds.length; index++) {
         const requestId = requestIds[index]
-        await withdrawalQueue.claimWithdrawal(requestId, { from: user })
+        await withdrawalQueue.claimWithdrawal(requestId, { from: user, gasPrice: 0 })
       }
       const balanceAfter = bn(await ethers.provider.getBalance(user))
       assert.equals(balanceAfter, balanceBefore.add(totalDistributedEth))
     })
-  })
-
-  context.skip('claim fuzzing', () => {
-    const fuzzClaim = async (perRequestWEI, requestCount, finalizedWEI) => {
-      await withdrawalQueue.requestWithdrawals(Array(requestCount).fill(perRequestWEI), user, { from: user })
-      const requestIds = await withdrawalQueue.getWithdrawalRequests(user, { from: user })
-
-      const id = await withdrawalQueue.getLastRequestId()
-      await withdrawalQueue.finalize([id], shareRate(1), { from: steth.address, value: finalizedWEI })
-
-      const hints = await withdrawalQueue.findCheckpointHints(
-        requestIds,
-        1,
-        await withdrawalQueue.getLastCheckpointIndex()
-      )
-
-      // this causes division by zero
-      const claimableEth = await withdrawalQueue.getClaimableEther(requestIds, hints).catch((e) => {
-        throw new Error(
-          // hack to fix error objects with bigInit causing `can't serialise bigInt` with wrong trace
-          JSON.parse(JSON.stringify(e, (_, value) => (typeof value === 'bigint' ? value.toString() : value)))
-        )
-      })
-
-      const totalClaimable = claimableEth.reduce((s, i) => s.iadd(i) && s, bn(0))
-      assert.equals(totalClaimable, finalizedWEI, `Total Claimable doesn't add up to finalized amount`)
 
+    it('100% discount', async () => {
       const balanceBefore = bn(await ethers.provider.getBalance(user))
-      await withdrawalQueue.claimWithdrawals(requestIds, hints, { from: user })
+      const id = await withdrawalQueue.getLastRequestId()
+      withdrawalQueue.finalize([id], 0, { from: steth.address, value: 0 })
```

Comment:
I'm not sure that it's a valid scenario, probably should be checked that value is not zero.

Reply:
I added prefinalize run , 1e-27 share rate and assert that total eth is 0. This extreme scenario tests that pre-finalization-claiming flow works even when there is none ether to give.